### PR TITLE
Scaled out StandByFeedIterator

### PR DESCRIPTION
**Pull request by ealsur**
## Description

Currently, the StandByFeedIterator is a single consumer pull model. This PR adds a new API called `GetChangeFeedTokensAsync`.

```c#
`Task<IEnumerable<string>> GetChangeFeedTokensAsync(CancellationToken cancellationToken = default(CancellationToken))`
```

The goal of this API is to provide a set of Tokens that can be used as initial state on any StandByFeedIterator to limit the scope of the sequential read to a subset of partitions.

This way, the tokens can be distributed on parallel compute units and each compute unit can read in parallel changes only for the scope of the initial token.

For example:

```c#
    IEnumerable<string> tokens = await container.GetChangeFeedTokensAsync();
    // In machine 1
    FeedIterator iteratorForToken =
                    container.GetStandByFeedIterator(continuationToken: tokens[0], requestOptions: new ChangeFeedRequestOptions() { StartTime = DateTime.MinValue });
    // In machine 2
    FeedIterator iteratorForToken =
                    container.GetStandByFeedIterator(continuationToken: tokens[1], requestOptions: new ChangeFeedRequestOptions() { StartTime = DateTime.MinValue });
``` 

Unit test and Emulator tests were added to cover the new feature.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
Please run this through this with Vinod. Bat.

File: Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs [hunk 1]
```diff
@@ -38,6 +38,15 @@ internal class StandByFeedContinuationToken
             return standByFeedContinuationToken;
         }
 
+        public static StandByFeedContinuationToken CreateForRange(
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs]: Please make the factory methods stateless as discussed offline.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs]: Done

File: Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs [hunk 2]
```diff
@@ -169,8 +195,8 @@ private void InitializeCompositeTokens(IEnumerable<CompositeContinuationToken> t
                 new Documents.Routing.Range<string>(
                     targetRange.Min,
                     targetRange.Max,
-                    isMaxInclusive: true,
-                    isMinInclusive: false),
+                    isMaxInclusive: false,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs]: Nice catch

File: Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs [hunk 3]
```diff
@@ -38,6 +38,28 @@ internal class StandByFeedContinuationToken
...
+            string maxExclusive)
+        {
+            if (string.IsNullOrWhiteSpace(containerRid)) throw new ArgumentNullException(nameof(containerRid));
+            // MinInclusive can be an empty string
+            if (minInclusive == null) throw new ArgumentNullException(nameof(minInclusive));
+            if (string.IsNullOrWhiteSpace(maxExclusive)) throw new ArgumentNullException(nameof(maxExclusive));
+            return StandByFeedContinuationToken.SerializeTokens(new CompositeContinuationToken[1] { StandByFeedContinuationToken.CreateCompositeContinuationTokenForRange(minInclusive, maxExclusive, null) });
+        }
+
+        public static string SerializeTokens(IEnumerable<CompositeContinuationToken> compositeContinuationTokens)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs]: Do these needs to be public?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Query/StandByFeedContinuationToken.cs]: Good catch

**Pull request merged**